### PR TITLE
Fix IVF_FLAT index's wrong meta_data

### PR DESCRIPTION
**Pull request by xige-16**
fix issue: #7128 #7077 #6797

Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

File: internal/core/unittest/test_index_wrapper.cpp [hunk 1]
```diff
@@ -278,6 +278,7 @@ TEST(IVFFLATNMWrapper, Build) {
 }
 
 TEST(IVFFLATNMWrapper, Codec) {
+    int64_t flat_nb = 1000000;
```
Comment (DragonDriver) [inline internal/core/unittest/test_index_wrapper.cpp:281]: 1000000 may be a very large number for ut. Consider moving this part to benchmark?

File: internal/core/src/index/knowhere/knowhere/common/Utils.cpp [hunk 2]
```diff
@@ -78,6 +78,15 @@ Assemble(BinarySet& binarySet) {
 void
 Disassemble(const int64_t& slice_size_in_byte, BinarySet& binarySet) {
     milvus::json meta_info;
+    auto slice_meta = binarySet.Erase(INDEX_FILE_SLICE_META);
+    if (slice_meta != nullptr) {
+        milvus::json last_meta_data =
+            milvus::json::parse(std::string(reinterpret_cast<char*>(slice_meta->data.get()), slice_meta->size));
+        for (auto& item : last_meta_data[META]) {
```
Comment (DragonDriver) [inline internal/core/src/index/knowhere/knowhere/common/Utils.cpp:85]: What if the item is duplicated with the later slice_i?

**Review by longjiquan — CHANGES_REQUESTED** [review]

File: internal/core/unittest/test_index_wrapper.cpp (hunk 1 above)
Comment (xige-16) [reply internal/core/unittest/test_index_wrapper.cpp:281]: Small data sets don't start with bugs.

File: internal/core/src/index/knowhere/knowhere/common/Utils.cpp (hunk 2 above)
Comment (xige-16) [reply internal/core/src/index/knowhere/knowhere/common/Utils.cpp:85]: It's impossible, size of data in last_meta_data has been smaller than slice_size_in_byte, and will not be duplicated with the later slice_i.

**Review by shengjun1985 — APPROVED** [review]
/lgtm

**Review by op-hunter — APPROVED** [review]

**Pull request merged**